march_hardware/src/IMotionCube.cpp

```diff
@@ -82,26 +82,26 @@ void IMotionCube::writeInitialSettings(uint8_t cycle_time)
   ROS_DEBUG("IMotionCube::writeInitialSettings");
 
   // mode of operation
-  int mode_of_op = sdo_bit8(slaveIndex, 0x6060, 0, this->actuation_mode_.toModeNumber());
+  int mode_of_op = sdo_bit8_write(slaveIndex, 0x6060, 0, this->actuation_mode_.toModeNumber());
 
   // position limit -- min position
-  int min_pos_lim = sdo_bit32(slaveIndex, 0x607D, 1, this->absolute_encoder_->getLowerSoftLimitIU());
+  int min_pos_lim = sdo_bit32_write(slaveIndex, 0x607D, 1, this->absolute_encoder_.getLowerSoftLimitIU());
```

Comment:
```suggestion
  int min_pos_lim = sdo_bit32_write(slaveIndex, 0x607D, 1, this->absolute_encoder_->getLowerSoftLimitIU());
```
absolute_encoder_ is a unique pointer (this is also why your build fails

Reply:
Trying to build this results in base operand of ‘->’ has non-pointer type ‘march::AbsoluteEncoder’ , the other notation does build on my system.